prov/tcp/src/tcpx_progress.c

```diff
@@ -113,7 +113,9 @@ static void process_rx_pe_entry(struct tcpx_pe_entry *pe_entry)
 done:
 	tcpx_cq_report_completion(pe_entry->ep->util_ep.rx_cq,
 				  pe_entry, ret);
-	dlist_remove(&pe_entry->entry);
+	if (!((pe_entry->entry.next == NULL) &&
+	      (pe_entry->entry.prev == NULL)))
```

Comment:
Why is this check needed?  This change looks unrelated and should be called out in the commit message.  (Have I mentioned adding more details to the commit messages?)  :)